### PR TITLE
[multi-asic][vs]: Add new multi-asic vs hwsku with four asics

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
**- Why I did it**
- Current mutli-asic vs hwsku consists of 6 asics with each asic having 32 interfaces. When bringing this up, below issue was seen:
- When all 32 interfaces(sonic interfaces and linux interface) are set to 9100 mtu, DMA error is seen "DMA: Out of SW-IOMMU space for 4096 bytes at device 0000:06:03.0" which can be fixed by updating swiotlb=65536 in /host/grub/grub.cfg .

In order to keep multi-asic VS lighter and easier to bring up and test, new hwsku 'msft_four_asic_vs' is added to represent 4-asic hwsku with 2 frontend asics and 2 backend asics and each asic having 8 interfaces interconnected by port-channels.

**- How I did it**
- Add msft_four_asic_hwsku directory to have the right number of directories (4) and update port_config.ini and lanemap.ini files to include 8 ports information.
- Add topology.sh script to create the internal asic-asic connectivity.

**- How to verify it**
- Update asic.conf with the 4 asic information as below and build sonic-vs.img:
NUM_ASIC=4
DEV_ID_ASIC_0=0
DEV_ID_ASIC_1=1
DEV_ID_ASIC_2=2
DEV_ID_ASIC_3=3
- Modify sonic_multiasic.xml to have 8 front panel interfaces.
- create virtual switch using "sudo virsh sonic_mutliasic.xml" command.
- Start topology service and Load config_db files for switch and each asic.
- Ensure that that all internal interfaces and port_channels are coming up.
multi-asic vs testbed:
- Bring up mutli-asic VS testbed with a multi-asic image(asic.conf updated to 4 asics) and using t1-lag topology.
 ./testbed-cli.sh -t vtestbed.csv -m veos_vtb -k ceos add-topo vms-kvm-four-asic-t1-lag password.txt
 - Load minigraph/config_dbs.
 - Ensure all internal and external interfaces come up.
- No change on single asic vs.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [X] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology_6_asic.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
```
Comment (arlakshm) [inline device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology_6_asic.sh:1]: I think we can remove this file

Comment (SuvarnaMeenakshi) [reply device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology_6_asic.sh]: Removed this file.

File: device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh [hunk 2]
```diff
@@ -1,73 +1,79 @@
...
+            echo "$ASIC: $OLD --> $NAME"
         done
     done
 
     for ASIC in `seq $FIRST_BACKEND_ASIC $LAST_BACKEND_ASIC`; do
-        for NUM in `seq 1 $NUM_INTERFACES_PER_ASIC`; do
+        for NUM in `seq 1 4`; do
             sudo ip netns exec asic$ASIC ip link set dev eth$NUM down
             sudo ip netns exec asic$ASIC ip link delete dev eth$NUM
+            echo "$ASIC: eth$NUM"
```
Comment (arlakshm) [inline device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh:76]: do we need this echo statement?

Comment (SuvarnaMeenakshi) [reply device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh]: We don't need this echo statement, removed echo statements added for debugging.

**Comment by judyjoseph** [comment]
LGTM

**Review by arlakshm — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]
Also to confirm the plan is to use the same file **sonic_multiasic.xml** for both 4 and 6 ASIC vs

**Comment by SuvarnaMeenakshi** [comment]
> Also to confirm the plan is to use the same file **sonic_multiasic.xml** for both 4 and 6 ASIC vs

sonic_mutliasic.xml in platform/vs directory supports 6 ASIC vs with 64 interfaces.
If we have to bring up 4 ASIC vs, this file has to be modified to have 8 external interfaces.
I have mentioned that point in the PR description, I will raise a separate PR to update README file to bring up 6-ASIC or 4-ASIC VS.

In sonic-mgmt, the xml file to create KVM is generated based on the number of front-panel interfaces.

**Comment by SuvarnaMeenakshi** [comment]
> > Also to confirm the plan is to use the file **sonic_multiasic.xml** for both 4 and 6 ASIC vs
> 
> sonic_mutliasic.xml in platform/vs directory supports 6 ASIC vs with 64 interfaces.
> If we have to bring up 4 ASIC vs, this file has to be modified to have 8 external interfaces.
> I have mentioned that point in the PR description, I will raise a separate PR to update README file to bring up 6-ASIC or 4-ASIC VS.
> 
> In sonic-mgmt, the xml file to create KVM is generated based on the number of front-panel interfaces.
> 
> Thanks, good to update the README.

Have updated README here: https://github.com/Azure/sonic-buildimage/pull/6867

**Pull request merged**